frontend/src/components/Products/index.tsx

```diff
@@ -1,26 +1,41 @@
 import { h } from 'preact';
 import { useEffect, useState } from 'preact/hooks';
+import { Link, useParams } from 'react-router-dom';
 import { Button } from '../Button';
 import { getProductsData } from '../../services/api';
 import { CartService } from '../../services/cartService';
+import slugMap from '../../services/slugMap.json';
```

Comment:
don't use service mock data directly in component
you should get id by slug via function